test/e2e/fishapp/dynamic_stack_test.go

```diff
@@ -0,0 +1,66 @@
+package fishapp_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/e2e/fishapp"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/e2e/fishapp/shared"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/e2e/framework"
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("tests by dynamic generated stack", func() {
+	var (
+		ctx          context.Context
+		f            *framework.Framework
+		dynamicStack *fishapp.DynamicStack
+	)
+
+	JustBeforeEach(func() {
+		ctx = context.Background()
+		f = framework.New(framework.GlobalOptions)
+		dynamicStack.Deploy(ctx, f)
+	})
+	JustAfterEach(func() {
+		dynamicStack.Cleanup(ctx, f)
+	})
+
+	Context("normal test dimensions with DNS", func() {
+		BeforeEach(func() {
+			dynamicStack = &fishapp.DynamicStack{
+				ServiceDiscoveryType:        shared.DNSServiceDiscovery,
+				VirtualServicesCount:        5,
+				VirtualNodesCount:           5,
+				RoutesCountPerVirtualRouter: 2,
+				TargetsCountPerRoute:        4,
+				BackendsCountPerVirtualNode: 2,
+				ReplicasPerVirtualNode:      3,
+				ConnectivityCheckPerURL:     100,
+			}
+		})
+		It("should behaves correctly", func() {
+			dynamicStack.ExpectBehavesCorrectly(ctx, f)
+		})
+	})
+
+	Context("normal test dimensions with cloudMap", func() {
+		BeforeEach(func() {
+			dynamicStack = &fishapp.DynamicStack{
+				ServiceDiscoveryType:        shared.CloudMapServiceDiscovery,
+				VirtualServicesCount:        5,
+				VirtualNodesCount:           5,
+				RoutesCountPerVirtualRouter: 2,
+				TargetsCountPerRoute:        4,
+				BackendsCountPerVirtualNode: 2,
+				ReplicasPerVirtualNode:      3,
+				ConnectivityCheckPerURL:     100,
```

Comment:
Non-blocking: I think it would be nice to be able to define these via a template or other argument format so they can be modified quickly and the tests run with different configurations.

Reply:
Thanks, i think it's a valuable idea. 
However, since tests is highly dynamic, and i think code itself is better than template :D. (e.g. we can have multiple tests tests different dimension combinations specified, currently i only specified one)